net/uradvd/src/uradvd.c

```diff
@@ -583,8 +588,30 @@ static void add_prefix(const char *prefix, bool adv_onlink) {
 
 static void parse_cmdline(int argc, char *argv[]) {
 	int c;
-	while ((c = getopt(argc, argv, "i:a:p:h")) != -1) {
+	char *endptr;
+	unsigned long val;
+
+	static struct option long_options[] =
+	{
+		{"default-lifetime", required_argument, 0, 0},
+		{0, 0, 0, 0}
+	};
+
+	int option_index = 0;
+
+	while ((c = getopt_long(argc, argv, "i:a:p:h", long_options, &option_index)) != -1) {
 		switch(c) {
+		case 0: // --default-lifetime
+			errno = 0;
+			val = strtoul(optarg, &endptr, 0);
+
+			if (((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN)) || (errno != 0 && val == 0) || endptr == optarg) && val <= UINT16_MAX)
```

Comment:
The `(val == LONG_MAX || val == LONG_MIN)` doesn't make sense for strtoul, and there are other mistakes in the conditoin. AFAICT, the following check would suffice:

    if (!*optarg || *endptr || val > UINT16_MAX)

*errno* can be ignored completely.